packages/core/android/src/main/kotlin/com/datadog/reactnative/VitalFrameCallback.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative
+
+import android.view.Choreographer
+import java.util.concurrent.TimeUnit
+
+/**
+ * Reads the javascript framerate based on the [Choreographer.FrameCallback].
+ */
+internal class VitalFrameCallback(
+    private val frameTimeCallback: (Double) -> Unit,
+    private val keepRunning: () -> Boolean
+) : Choreographer.FrameCallback {
+
+    internal var lastFrameTimestampNs: Long = 0L
+
+    // region Choreographer.FrameCallback
+
+    override fun doFrame(frameTimeNanos: Long) {
+        if (lastFrameTimestampNs != 0L) {
+            val durationNs = (frameTimeNanos - lastFrameTimestampNs).toDouble()
+            frameTimeCallback(durationNs)
+        }
+        lastFrameTimestampNs = frameTimeNanos
+
+        if (keepRunning()) {
+            try {
+                Choreographer.getInstance().postFrameCallback(this)
+            } catch (e: IllegalStateException) {
+                // TODO: log error here
```

Comment:
seems like unfinished TODO here

Reply:
I had a previous comment in the PR for this TODO, and I think we could remove it completely:

I'm not sure it's worth logging this error.
It would require to inject a error handler into the VitalFrameCallback, making the code more complex that I think it should.
I think in theory it should not happen if we've been able to start the Frame Callback from the SDK where the error is handled.

If you think I should log the error I will pass a error handler to the VitalFrameCallback contructor.

Reply:
imo it is still useful to log at least locally for the SDK dev builds. Even things which should never happen can happen due to various reasons (not necessarily because of the issue on our side).